### DISCUSSION TITLE
Support for mathematics in discussions

### DISCUSSION THREAD

**Discussion by prudhomm** [community/community#2816] [Discussions]
Our OSS repo does not have Github Discussions (we would be very happy to have it enabled though :) ).
We are developing software using a lot of math. A. lot of the questions we have are related to the mathematics. Would it be possible to support math (à la latex) in markdown or asciidoc ? That would an absolutely awesome addition for many projects hosted at Github

**Comment by becca** [comment]
👋 thank you for sharing this feedback @prudhomm. I'm hopeful we can help your community use Discussions on GitHub.

Do you have examples of where these questions in your community currently live?

**Reply by prudhomm** [reply]
We actually don't do that. We use Slack and Slack does not support math. The mathematical language is used in many aspects of engineering and research and being able to use it would help convey better(in a simpler way) our meanings.
Currently when needed we use directly latex like
```
-\Delta u = f \mbox{ on } \Omega, u=g \mbox{ on } \partial Omega
```
but it is better like this (in png below) (using mathjax it would be much better)

![image](https://user-images.githubusercontent.com/151693/101366950-e2c12480-38a5-11eb-861b-03323c0b14f8.png)

Gitter (gitter.im) has latex support and stackoverflow as well.

**Reply by paolieri** [reply]
This would be amazing to have in GitHub discussions.

Discourse also has support for MathJax through plugins:
https://meta.discourse.org/t/discourse-math-plugin/65770

**Reply by ljq** [reply]
It is recommended to add some simple support

**Reply by emstoudenmire** [reply]
Hi @becca, I'd really appreciate Github Discussions have this feature too. It would be a real selling point for us.

As examples of where and how our community posts mathematics as part of questions, here are two:
- General Julia Language Discourse forum, for example this post: https://discourse.julialang.org/t/solving-pde-diffusion-using-ode-solvers-total-mass-increases-towards-infinity-should-stay-constant/67492
- Our own software (ITensor)'s very ugly :^) but effective forum: 
  * Example 1: http://itensor.org/support/3350/calculating-observables-from-density-matrices
  * Example 2: http://itensor.org/support/3295/how-to-construct-bose-hubbard-model

We've been wanting to upgrade our forum, and just learned about Github Discussions, but we'd also like to keep the ability of our users to write mathematical expressions in their questions.

**Comment by deniscostadsc** [comment]
This subject has a long discussion[1], and some people[2] (me included) think that, although this is not a very popular request, Github should implement it.

It seems Github had support math equations rendering in markdown in the past, and had some trouble maintaining this[3]. But that testimonial was from 7 years ago, and I'm quite sure markdown parsing and rendering has evolved a lot since than.

Gitlab implements math equations rendering in markdown[4], and they seem to use Kramdown[5]. I'm not suggesting you just replace Github Flavored Markdown for Gitlab's one, but I think I worth take a look on how they did it.

[1] - https://github.com/github/markup/issues/274
[2] - https://stackoverflow.com/a/61251740
[3] - https://github.com/github/markup/issues/274#issuecomment-39338272
[4] - https://docs.gitlab.com/ee/user/markdown.html#math
[5] - https://kramdown.gettalong.org/

**Reply by tuliocasagrande** [reply]
I don't really understand the rendering engines, but just to add that Jupyter Notebooks [1] uses MathJax [2].

[1] - https://nbviewer.jupyter.org/github/ipython/ipython/blob/2.x/examples/Notebook/Display%20System.ipynb#LaTeX
[2] - https://www.mathjax.org/

**Reply by Andre601** [reply]
Did you try to use the reference link format?
Because that format is `[1]: <URL>` as source and `[1]` for where you want to implement it.

More info I made: https://www.andre601.ch/GitHub-Markdown/github/general.html#reference

**Reply by deniscostadsc** [reply]
I know I can use external sources to render LaTeX, like Codecogs, and other. But that make markdown editing/reading more difficult and error prone.

Instead of using:

```
![](https://latex.codecogs.com/svg.latex?\Large&space;x=\frac{-b\pm\sqrt{b^2-4ac}}{2a})
```

for render

![](https://latex.codecogs.com/svg.latex?\Large&space;x=\frac{-b\pm\sqrt{b^2-4ac}}{2a})

_As you can see above, sometimes external service don't render nicely. At least the image above is note being rendered sometimes for me, but you can click on it to view it._

It seems to be much easier to write something like this:

 ```
$x=\frac{-b\pm\sqrt{b^2-4ac}}{2a}$
```

**Reply by patel-zeel** [reply]
I strongly support that latex-type math support would be extremely useful in GitHub discussions as well as other places (pull requests, issues). It has been so long since I am trying to find out another alternative for math-supported platforms that support GitHub discussions like applications but couldn't find a single sustainable solution. I just hope GitHub starts supporting it soon.

**Comment by nschloe** [comment]
For the time being, the [xhub](https://github.com/nschloe/xhub) browser extension may serve as a drop-in replacement for proper math support. (Has disadvantages: Doesn't work on mobile, Chrome-only etc.)

**Comment by davidstaheli** [comment]
Thanks to everyone for posting about this feature request and upvoting it. GitHub is actively working on it now.

**Reply by heykapil** [reply]
Yes please! It'llbe awesome :)

**Comment by davidstaheli** [comment]
Support for mathematical expressions in Markdown (and discussions) is in progress.

**Reply by BurhanH** [reply]
Finally! - https://github.blog/2022-05-19-math-support-in-markdown/

**Reply by laymonage** [reply]
Hey, any chance to add $\LaTeX$ support in the [Markdown API](https://docs.github.com/en/rest/markdown) as well? Thanks!

**Reply by fjp** [reply]
Are there also plans to support this in AsciiDoc?

**Comment by nschloe** [comment]
So this has been implemented and released a few days ago. Before everyone gets too excited about it: There are some fundamental flaws on how it's done that you might want to be aware of. See [my blog post](https://nschloe.github.io/2022/05/20/math-on-github.html) or [GitHub's own feedback page](https://github.com/github/feedback/discussions?discussions_q=math).

**Comment by Hamzamalikseo** [comment]
This is a really useful idea. Having better math support in Markdown (like LaTeX rendering) would make explanations much clearer, especially for students trying to learn step-by-step problem solving. Right now, many people struggle to express formulas properly in text-based discussions. In learning situations, breaking [calculations into steps](https://lacalculadora-dealicia.com/) is very important for understanding the logic behind the answer. Tools like calculadora de alicia can also help students by showing step-by-step solutions for math problems, making the learning process easier and more structured.
